File: Example/ObjC(Deprecated)/checkout-demo-objc/Checkout/Cells/CheckoutSummaryTableViewCell.m
//
//  CheckoutSummaryTableViewCell.m
//  athm-checkout-demo-objc
//
//  Created by Cristopher Bautista on 7/10/18.
//  Copyright © 2018 EVERTEC. All rights reserved.
//

#import "CheckoutSummaryTableViewCell.h"
#import "Transaction.h"

@interface CheckoutSummaryTableViewCell()
//@property(nonatomic,retain)Transaction * transaction;
@end
@implementation CheckoutSummaryTableViewCell
@synthesize contentView, totalLabel, totalValueLabel, subTotalValueLabel;
@synthesize subTotalLabel, taxLabel, taxValueLabel ;//, transaction;
+(NSString *)reuseIdentifier {
    return @"CheckoutSummaryTableViewCell";
}
- (void)awakeFromNib {
    [super awakeFromNib];
    // Initialization code
    contentView.backgroundColor = [UIColor whiteColor];
    totalLabel.textColor = [UIColor blueColor];
    totalValueLabel.textColor = [UIColor blueColor];
    totalLabel.font = [UIFont boldSystemFontOfSize:18];
    totalValueLabel.font = [UIFont boldSystemFontOfSize:18];
}

-(void)configWith:(Transaction *)transaction {
    subTotalValueLabel.text = transaction.subTotal;
    taxValueLabel.text = transaction.tax;
    totalValueLabel.text = transaction.total;
    subTotalLabel.text = @"Subtotal";
    taxLabel.text = @"Tax(12%)";
    totalLabel.text = @"TOTAL";
}

- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
    [super setSelected:selected animated:animated];

    // Configure the view for the selected state
}

@end

File: Example/athmovil-checkout/Checkout/CheckoutSummaryTableViewCell.swift
//
//  CheckoutSummaryTableViewCell.swift
//  athm-checkout-demo
//
//  Created by Leonardo Maldonado on 5/4/18.
//  Copyright © 2018 Leonardo Maldonado. All rights reserved.
//

import UIKit

class CheckoutSummaryTableViewCell: UITableViewCell {
    
    static let reuseIdentifier: String = "CheckoutSummaryTableViewCell"

    @IBOutlet weak var subTotalLabel: UILabel!
    @IBOutlet weak var taxLabel: UILabel!
    @IBOutlet weak var totalLabel: UILabel!
    @IBOutlet weak var subTotalValueLabel: UILabel!
    @IBOutlet weak var taxValueLabel: UILabel!
    @IBOutlet weak var totalValueLabel: UILabel!
    
    var transaction: Transaction? {
        didSet {
            subTotalValueLabel.text = transaction?.subTotal
            taxValueLabel.text = transaction?.tax
            totalValueLabel.text = transaction?.total
            subTotalLabel.text = "Subtotal"
            taxLabel.text = "Tax(12%)"
            totalLabel.text = "TOTAL"
        }
    }
    
    override func awakeFromNib() {
        super.awakeFromNib()
        
        contentView.backgroundColor = .white
        totalLabel.textColor = .default
        totalValueLabel.textColor = .default
        totalLabel.font = UIFont.boldSystemFont(ofSize: 18)
        totalValueLabel.font = UIFont.boldSystemFont(ofSize: 18)
    }
}
